src/charm.py

```diff
@@ -242,23 +248,23 @@ def _get_certificate_requests(self) -> typing.List[CertificateRequestAttributes]
             )
         ]
 
-    @validate_config_and_tls(defer=True, block_on_tls_not_ready=True)
+    @validate_config_and_tls(defer=True)
     def _on_ingress_data_provided(self, event: IngressPerAppDataProvidedEvent) -> None:
         """Handle the data-provided event.
 
         Args:
             event: Juju event.
         """
         self._reconcile()
-        tls_information = TLSInformation.from_charm(self, self.certificates)
-        integration_data = self._ingress_provider.get_data(event.relation)
-        path_prefix = f"{integration_data.app.model}-{integration_data.app.name}"
-
-        self._ingress_provider.publish_url(
-            event.relation, f"https://{tls_information.external_hostname}/{path_prefix}/"
-        )
+        if self.unit.is_leader():
+            tls_information = TLSInformation.from_charm(self, self.certificates)
```

Comment:
This is already available in a charm attr. See https://github.com/canonical/haproxy-operator/blob/8e83e8c6cdf4f38cd9e50b800d8621365007ad4d/src/charm.py#L194

Reply:
Nvm, this is cool :)